src/server/game/AI/SmartScripts/SmartScriptMgr.h

```diff
@@ -1240,7 +1240,10 @@ enum SMARTAI_TARGETS
     SMART_TARGET_FARTHEST                       = 28,   // maxDist, playerOnly, isInLos
     SMART_TARGET_VEHICLE_PASSENGER              = 29,   // TODO: NOT SUPPORTED YET
 
-    SMART_TARGET_END                            = 30
+    SMART_TARGET_PLAYER_WITH_AURA               = 50,   // spellId, negation, MaxDist, MinDist, set target.o to a number to random resize the list
+    SMART_TARGET_RANDOM_POINT                   = 51,   // range, amount (for summoning creature), self als middle (0/1) else use xyz
+
+    SMART_TARGET_END                            = 52
```

Comment:
we need a way to differentiate AC-only targets, like we did with Actions:

https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/AI/SmartScripts/SmartScriptMgr.h#L571